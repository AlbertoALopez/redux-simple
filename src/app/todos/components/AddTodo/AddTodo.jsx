import React from 'react';

const AddTodo = ({ handleTodoAdd }) => {
    let input;
    return (
        <div>
            <input
                ref={(node) => {
                    input = node;
                }}
            />
            <button
                onClick={() => {
                    handleTodoAdd(input.value);
                }}
            >
                Add Todo
            </button>
        </div>
    );
};

export default AddTodo;
