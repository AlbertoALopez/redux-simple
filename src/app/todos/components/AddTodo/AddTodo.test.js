import React from 'react';
import AddTodo from './AddTodo.jsx';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

function handleAddToDo(value) {
    console.log(value);
}

test('AddTodo should render', () => {
    const component = renderer.create(
        <AddTodo
            onClick={handleAddToDo.bind(this)}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('AddTodo should contain a button and input field', () => {
    const wrapper = mount(
        <AddTodo
            onClick={handleAddToDo.bind(this)}
        />
    );

    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('input').length).toBe(1);
});
