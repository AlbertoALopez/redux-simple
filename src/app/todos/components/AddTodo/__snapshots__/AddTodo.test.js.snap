exports[`test AddTodo should render 1`] = `
<div>
  <input />
  <button
    onClick={[Function]}>
    Add Todo
  </button>
</div>
`;
